### ISSUE TITLE
MueLu: MueLu_ExplicitInstantiation.hpp missing in non-eti build.

### ISSUE THREAD

**Issue by william76** [trilinos/Trilinos#6419]
@trilinos/muelu 

### Description

This is coming from the error @ndellingwood and I see in the build from #6414 where his build doesn't enable ETI.  We let that PR go through since it fixes some other circular reference issues in Xpetra but this issue was still in place and looks like it's something in MueLu.

With ETI disabled, MueLu can fail to build with the following error:
```
In file included from /ascldap/users/wcmclen/dev/trilinos-dev/build/xpetra-eti-issue-6314/packages/muelu/src/MueLu_RefMaxwell.hpp:1:0,
                 from /ascldap/users/wcmclen/dev/trilinos-dev/source/Trilinos/packages/muelu/adapters/aztecoo/MueLu_AztecEpetraOperator.cpp:8:
/ascldap/users/wcmclen/dev/trilinos-dev/source/Trilinos/packages/muelu/adapters/xpetra/MueLu_RefMaxwell_decl.hpp:50:43: fatal error: MueLu_ExplicitInstantiation.hpp: No such file or directory
 #include "MueLu_ExplicitInstantiation.hpp"
                                           ^
compilation terminated.
make[2]: *** [packages/muelu/adapters/CMakeFiles/muelu-adapters.dir/aztecoo/MueLu_AztecEpetraOperator.cpp.o] Error 1
```
If I turn on `Trilinos_EXPLICIT_INSTANTIATION` then I don't get this error. There probably needs to be some guard in `MueLu_AztecEpetraOperator` for ETI.  Also, should ETI just be turned on by default in CMake?  Is there anyone we know of that _needs_ ETI to be off?

#### Set Environment
```bash
module purge
module load sems-env
module load sems-devpack-gcc/6.1.0
module load sems-cmake/3.12.2
module load sems-git/2.10.1
```
#### Configuration
- [do-configure.bash.txt](https://github.com/trilinos/Trilinos/files/3945938/do-configure.bash.txt) - Change `TRILINOS_PATH` to be the path to where you have Trilinos cloned.
#### Outputs
- [configure.log](https://github.com/trilinos/Trilinos/files/3945940/__output.configure.log)
- [make.log](https://github.com/trilinos/Trilinos/files/3945941/__output.make.log)

**Comment by cgcgcg** [comment]
@ndellingwood Could you try with the latest develop?

**Comment by ndellingwood** [comment]
@cgcgcg build in progress, pretty slow on my laptop so may be mid-afternoon before I get back to you with results.

**Comment by ndellingwood** [comment]
@cgcgcg my build makes it past the RefMaxwell error addressed, now it is failing in `Xpetra_MapExtractor` with errors like

```
In file included from /Users/ndellin/Research/trilinos/Trilinos-stable/Build/Build-Pristine/packages/xpetra/src/Xpetra_MapExtractor.hpp:2:0,
                 from /Users/ndellin/Research/trilinos/Trilinos-stable/packages/xpetra/src/BlockedMultiVector/Xpetra_BlockedMultiVector_def.hpp:54,
                 from /Users/ndellin/Research/trilinos/Trilinos-stable/Build/Build-Pristine/packages/xpetra/src/Xpetra_BlockedMultiVector.hpp:2,
                 from /Users/ndellin/Research/trilinos/Trilinos-stable/packages/xpetra/src/MultiVector/Xpetra_EpetraMultiVectorFactory.cpp:47:
/Users/ndellin/Research/trilinos/Trilinos-stable/packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractor_def.hpp: In member function 'Teuchos::RCP<Xpetra::   Vector<Scalar, LocalOrdinal, GlobalOrdinal, Node> > Xpetra::MapExtractor<S, LO, GO, N>::ExtractVector(Teuchos::RCP<Xpetra::Vector<Scalar, LocalOrdinal,        GlobalOrdinal, Node> >&, size_t, bool) const':

/Users/ndellin/Research/trilinos/Trilinos-stable/packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractor_def.hpp:197:32: error: 'template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node> class Xpetra::VectorFactory' used without template parameters
         const RCP<Vector> vv = VectorFactory::Build(getMap(block, false), false);
                                ^~~~~~~~~~~~~
/Users/ndellin/Research/trilinos/Trilinos-stable/packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractor_def.hpp: In member function 'Teuchos::RCP<Xpetra::   Vector<Scalar, LocalOrdinal, GlobalOrdinal, Node> > Xpetra::MapExtractor<S, LO, GO, N>::getVector(size_t, bool, bool) const':
/Users/ndellin/Research/trilinos/Trilinos-stable/packages/xpetra/src/BlockedCrsMatrix/Xpetra_MapExtractor_def.hpp:569:16: error: 'template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node> class Xpetra::VectorFactory' used without template parameters
         return VectorFactory::Build(getMap(i, bThyraMode), bZero);
```

**Comment by ndellingwood** [comment]
@cgcgcg @william76 the specific item in this issue has been resolved, should I close? I think #6322 is tracking the error I posted above.

**Comment by william76** [comment]
@ndellingwood fine with me.
